--- /home/toolchain/development/libc++_replacement/tmp/GBS-ROOT/LIBCXX-2218-aarch64-20260806-c2/local/BUILD-ROOTS/scratch.aarch64.0/usr/bin/ldd	2026-08-05 00:12:36.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/progress/R9C/seg2/probes/ldd.aarch64.qemu-static	2026-08-06 10:43:57.733523912 +0800
@@ -26,7 +26,7 @@
 TEXTDOMAIN=libc
 TEXTDOMAINDIR=/usr/share/locale
 
-RTLDLIST=/lib64/ld-linux-aarch64.so.1
+RTLDLIST=/home/toolchain/development/libc++_replacement/progress/R9C/seg2/probes/rtld-aarch64-qemu-wrapper.bin
 warn=
 bind_now=
 verbose=
