--- /home/toolchain/development/libc++_replacement/tmp/GBS-ROOT/LIBCXX-2218-armv7l-20260806-c2/local/BUILD-ROOTS/scratch.armv7l.0/usr/bin/ldd	2026-08-04 23:41:34.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/progress/R9C/seg2/probes/ldd.armv7l.qemu-static	2026-08-06 10:43:57.550968063 +0800
@@ -26,7 +26,7 @@
 TEXTDOMAIN=libc
 TEXTDOMAINDIR=/usr/share/locale
 
-RTLDLIST=/lib/ld-linux.so.3
+RTLDLIST=/home/toolchain/development/libc++_replacement/progress/R9C/seg2/probes/rtld-armv7l-qemu-wrapper.bin
 warn=
 bind_now=
 verbose=
