--- /home/toolchain/development/libc++_replacement/tmp/corpus/extracted/llvm-22.1.8-18.1.src.rpm/unpacked/llvm-22.1.8.tar.gz/llvm-22.1.8/clang/lib/Driver/ToolChains/CommonArgs.cpp	2026-06-16 06:33:53.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/codes/llvm/clang/lib/Driver/ToolChains/CommonArgs.cpp	2026-08-05 11:41:55.726448894 +0800
@@ -1029,7 +1029,7 @@
 #endif
 
     SmallString<1024> Plugin;
-    llvm::sys::path::native(Twine(D.Dir) +
+    llvm::sys::path::native(Twine(D.RealDir) +
                                 "/../" CLANG_INSTALL_LIBDIR_BASENAME +
                                 PluginName + Suffix,
                             Plugin);
