--- /home/toolchain/development/libc++_replacement/tmp/R8/platform_srpm/patch_applied_subset/clang/lib/Options/OptionUtils.cpp	2026-08-05 16:40:10.937507991 +0800
+++ /home/toolchain/development/libc++_replacement/codes/llvm/clang/lib/Options/OptionUtils.cpp	2026-08-05 11:41:55.810546450 +0800
@@ -15,6 +15,7 @@
 #include "llvm/Option/ArgList.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/Path.h"
+#include <cstdlib> // ::getenv
 
 using namespace clang;
 using namespace llvm::opt;
@@ -228,7 +229,7 @@
     // On non-Windows, libclang.so/.dylib is in lib/.
     // With a static-library build of libclang, LibClangPath will contain the
     // path of the embedding binary, which for LLVM binaries will be in bin/.
-    // ../lib gets us to lib/ in both cases.c
+    // ../lib gets us to lib/ in both cases.
     const char *NO_LIBDIR_SUFFIX = std::getenv("CLANG_NO_LIBDIR_SUFFIX");
     StringRef ClangLibdirSuffix(NO_LIBDIR_SUFFIX ? "" : CLANG_LIBDIR_SUFFIX);
     P = llvm::sys::path::parent_path(Dir);
