--- /home/toolchain/development/libc++_replacement/tmp/corpus/extracted/llvm-22.1.8-18.1.src.rpm/unpacked/llvm-22.1.8.tar.gz/llvm-22.1.8/clang/lib/Driver/Driver.cpp	2026-06-16 06:33:53.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/codes/llvm/clang/lib/Driver/Driver.cpp	2026-08-05 11:41:55.719841931 +0800
@@ -186,6 +186,13 @@
   if (!this->VFS)
     this->VFS = llvm::vfs::getRealFileSystem();
 
+  RealDir = llvm::sys::path::parent_path(ClangExecutable);
+
+// Skip /emul if we work in accelerated environment
+#define QEMU_ACCEL_DIR "/emul"
+  if (ClangExecutable.substr(0, sizeof(QEMU_ACCEL_DIR) - 1) == QEMU_ACCEL_DIR)
+    ClangExecutable = ClangExecutable.drop_front(sizeof(QEMU_ACCEL_DIR) - 1);
+
   Name = std::string(llvm::sys::path::filename(ClangExecutable));
   Dir = std::string(llvm::sys::path::parent_path(ClangExecutable));
 
