--- /home/toolchain/development/libc++_replacement/tmp/R8/platform_srpm/patch_applied_subset/llvm/lib/TargetParser/Triple.cpp	2026-08-05 16:40:10.937976918 +0800
+++ /home/toolchain/development/libc++_replacement/codes/llvm/llvm/lib/TargetParser/Triple.cpp	2026-08-05 11:41:59.109887954 +0800
@@ -283,6 +283,8 @@
   case SUSE: return "suse";
   case Meta:
     return "meta";
+  case Tizen:
+    return "tizen";
   }
 
   llvm_unreachable("Invalid VendorType!");
