--- /home/toolchain/development/libc++_replacement/tmp/corpus/extracted/llvm-22.1.8-18.1.src.rpm/unpacked/llvm-22.1.8.tar.gz/llvm-22.1.8/llvm/lib/TargetParser/Triple.cpp	2026-06-16 06:33:53.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/codes/llvm/llvm/lib/TargetParser/Triple.cpp	2026-08-05 11:41:59.109887954 +0800
@@ -283,6 +283,8 @@
   case SUSE: return "suse";
   case Meta:
     return "meta";
+  case Tizen:
+    return "tizen";
   }
 
   llvm_unreachable("Invalid VendorType!");
@@ -703,6 +705,7 @@
       .Case("oe", Triple::OpenEmbedded)
       .Case("intel", Triple::Intel)
       .Case("meta", Triple::Meta)
+      .Case("tizen", Triple::Tizen)
       .Default(Triple::UnknownVendor);
 }
 
