--- progress/R68/tools/sdb_executor.before_utf8_fix.py	2026-08-27 13:42:54.100911372 +0800
+++ progress/R68/tools/sdb_executor.after_utf8_fix.py	2026-08-27 13:42:50.322202826 +0800
@@ -121,6 +121,7 @@
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             text=True,
+            errors="replace",
         )
         telemetry_path = os.environ.get("R68_SDB_TELEMETRY")
         if telemetry_path:
