--- tmp/R54/source-all/libcxx/include/future	2026-08-21 12:46:52.167007773 +0800
+++ tmp/R55/source-fixed/libcxx/include/future	2026-08-21 14:02:31.635472752 +0800
@@ -856,6 +856,11 @@
     this->set_value(__func_());
 #    if _LIBCPP_HAS_EXCEPTIONS
   } catch (const __cxxabiv1::__forced_unwind&) {
+    try {
+      std::__throw_future_error(future_errc::broken_promise);
+    } catch (...) {
+      this->set_exception(current_exception());
+    }
     throw;
   } catch (...) {
     this->set_exception(current_exception());
@@ -895,6 +900,11 @@
     this->set_value();
 #    if _LIBCPP_HAS_EXCEPTIONS
   } catch (const __cxxabiv1::__forced_unwind&) {
+    try {
+      std::__throw_future_error(future_errc::broken_promise);
+    } catch (...) {
+      this->set_exception(current_exception());
+    }
     throw;
   } catch (...) {
     this->set_exception(current_exception());
DIFF_EXIT_CODE=1
