diff --git a/tmp/R33/llvm-exp/libcxxabi/src/cxa_exception.cpp b/tmp/R34B/llvm-exp/libcxxabi/src/cxa_exception.cpp
index 5d7edae..c7d135e 100644
--- a/tmp/R33/llvm-exp/libcxxabi/src/cxa_exception.cpp
+++ b/tmp/R34B/llvm-exp/libcxxabi/src/cxa_exception.cpp
@@ -294,7 +294,10 @@ __cxa_throw(void *thrown_object, std::type_info *tinfo, void (_LIBCXXABI_DTOR_FU
 #ifdef __USING_SJLJ_EXCEPTIONS__
     _Unwind_SjLj_RaiseException(&exception_header->unwindHeader);
 #else
-    _Unwind_RaiseException(&exception_header->unwindHeader);
+    if (native_exception)
+        _Unwind_RaiseException(&exception_header->unwindHeader);
+    else
+        _Unwind_Resume_or_Rethrow(&exception_header->unwindHeader);
 #endif
     //  This only happens when there is no handler, or some unexpected unwinding
     //     error happens.
