diff --git a/tmp/R33/llvm-exp/libcxxabi/src/cxa_exception.cpp b/tmp/R34B/llvm-exp/libcxxabi/src/cxa_exception.cpp
index 5d7edae..0a346f3 100644
--- a/tmp/R33/llvm-exp/libcxxabi/src/cxa_exception.cpp
+++ b/tmp/R34B/llvm-exp/libcxxabi/src/cxa_exception.cpp
@@ -648,7 +648,10 @@ void __cxa_rethrow() {
 #ifdef __USING_SJLJ_EXCEPTIONS__
     _Unwind_SjLj_RaiseException(&exception_header->unwindHeader);
 #else
-    _Unwind_RaiseException(&exception_header->unwindHeader);
+    if (native_exception)
+        _Unwind_RaiseException(&exception_header->unwindHeader);
+    else
+        _Unwind_Resume_or_Rethrow(&exception_header->unwindHeader);
 #endif
 
     //  If we get here, some kind of unwinding error has occurred.
