--- tmp/R52/source-layer-c/libcxx/include/__ostream/basic_ostream.h	2026-08-05 11:41:57.672754264 +0800
+++ tmp/R53/source-layer-c-full/libcxx/include/__ostream/basic_ostream.h	2026-08-20 22:50:14.721301183 +0800
@@ -13,6 +13,7 @@
 
 #if _LIBCPP_HAS_LOCALIZATION
 
+#  include <cxxabi.h>
 #  include <__exception/operations.h>
 #  include <__fwd/memory.h>
 #  include <__iterator/ostreambuf_iterator.h>
@@ -214,6 +215,8 @@
       if (__os_.rdbuf()->pubsync() == -1)
         __os_.setstate(ios_base::badbit);
 #  if _LIBCPP_HAS_EXCEPTIONS
+    } catch (const __cxxabiv1::__forced_unwind&) {
+      throw;
     } catch (...) {
     }
 #  endif // _LIBCPP_HAS_EXCEPTIONS
