--- codes/llvm/libcxx/include/istream	2026-08-05 11:41:57.689213251 +0800
+++ tmp/R46/experiment/libcxx/include/istream	2026-08-17 13:49:52.955747497 +0800
@@ -166,6 +166,7 @@
 
 #  if _LIBCPP_HAS_LOCALIZATION
 
+#    include <cxxabi.h>
 #    include <__fwd/istream.h>
 #    include <__iterator/istreambuf_iterator.h>
 #    include <__locale_dir/num.h>
@@ -1325,6 +1326,10 @@
       __state |= ios_base::eofbit | (__str.empty() ? ios_base::failbit : ios_base::goodbit);
 
 #    if _LIBCPP_HAS_EXCEPTIONS
+  } catch (const __cxxabiv1::__forced_unwind&) {
+    __state |= ios_base::badbit;
+    __is.__setstate_nothrow(__state);
+    throw;
   } catch (...) {
     __state |= ios_base::badbit;
     __is.__setstate_nothrow(__state);
