--- tmp/BUILD_W1_0908/interface/old_x86/InlinerSizeModel.h	2026-06-29 14:20:10.000000000 +0800
+++ tmp/BUILD_W1_0908/interface/new_x86/InlinerSizeModel.h	2026-09-07 22:18:53.380688700 +0800
@@ -6,8 +6,8 @@
 //
 // clang-format off
 
-#ifndef TFCOMPILE_GENERATED__xla_home_linhao_mlgo_assets_x86_64_publish_InlinerSizeModel_llvm_InlinerSizeModel_H_  // NOLINT(build/header_guard)
-#define TFCOMPILE_GENERATED__xla_home_linhao_mlgo_assets_x86_64_publish_InlinerSizeModel_llvm_InlinerSizeModel_H_  // NOLINT(build/header_guard)
+#ifndef TFCOMPILE_GENERATED__xla_InlinerSizeModel_llvm_InlinerSizeModel_H_  // NOLINT(build/header_guard)
+#define TFCOMPILE_GENERATED__xla_InlinerSizeModel_llvm_InlinerSizeModel_H_  // NOLINT(build/header_guard)
 
 
 
@@ -18,7 +18,7 @@
 namespace xla { class ExecutableRunOptions; }
 
 // (Implementation detail) Entry point to the function in the object file.
-extern "C" void _xla_home_linhao_mlgo_assets_x86_64_publish_InlinerSizeModel_llvm_InlinerSizeModel(
+extern "C" void _xla_InlinerSizeModel_llvm_InlinerSizeModel(
     void* result, const ::xla::ExecutableRunOptions* run_options,
     const void** args, void** temps, XlaCustomCallStatus* status,
     int64_t* profile_counters);
@@ -79,7 +79,7 @@
     static XlaCompiledCpuFunction::StaticData* kStaticData = [](){
       XlaCompiledCpuFunction::StaticData* data =
         new XlaCompiledCpuFunction::StaticData;
-      set_static_data_raw_function(data, _xla_home_linhao_mlgo_assets_x86_64_publish_InlinerSizeModel_llvm_InlinerSizeModel);
+      set_static_data_raw_function(data, _xla_InlinerSizeModel_llvm_InlinerSizeModel);
       set_static_data_buffer_infos(data, BufferInfos());
       set_static_data_num_buffers(data, kNumBuffers);
       set_static_data_result_index_table(data, ResultIndexToBufferIndex());
@@ -96,7 +96,6 @@
       set_static_data_program_shape(data, StaticProgramShape());
       set_static_data_hlo_profile_printer_data(
           data, StaticHloProfilePrinterData());
-      set_static_data_use_xla_runtime(data, false);
 
       return data;
     }();
@@ -2270,6 +2269,6 @@
 
 }  // end namespace llvm
 
-#endif  // TFCOMPILE_GENERATED__xla_home_linhao_mlgo_assets_x86_64_publish_InlinerSizeModel_llvm_InlinerSizeModel_H_
+#endif  // TFCOMPILE_GENERATED__xla_InlinerSizeModel_llvm_InlinerSizeModel_H_
 
 // clang-format on
