--- codes/qb/TIZEN/Tizen/Tizen-Base-Toolchain/project_config
+++ codes/qb/TIZEN/Tizen/Tizen-Base-Toolchain/project_config
@@ -288,6 +288,7 @@
 Support: which  elfutils
 Support: update-alternatives
 Support: libstdc++-devel
+Support: libc++-devel libc++abi-devel
 Support: cpp
 Support: libatomic
 Support: libgomp
@@ -296,7 +297,9 @@
 Support: app-rootstrap-checker
 
 Keep: libstdc++-devel
+Keep: libc++-devel libc++abi-devel
 Keep: cpp gcc libstdc++
+Keep: libc++ libc++abi
 Keep: pam
 Keep: binutils cpp  libmagic-data file findutils gawk gcc  gcc-c++
 Keep: gdbm gzip libada libunwind  glibc-devel libpcre xz-lzma-compat
