--- codes/qb/TIZEN/Tizen/Tizen-Unified-Toolchain/project_config
+++ codes/qb/TIZEN/Tizen/Tizen-Unified-Toolchain/project_config
@@ -153,6 +153,8 @@
 export AR=%{__ar} RANLIB=%{__ranlib};\
 export CC=%{__cc} CXX=%{__cxx};\
 export MAKEFLAGS+=" -k";\
+export CXXFLAGS="${CXXFLAGS} -stdlib=libc++";\
+export LDFLAGS="${LDFLAGS} -stdlib=libc++ -Wl,--no-as-needed -lc++abi -Wl,--as-needed";\
 %{nil}
 :Macros
 
