--- tmp/R52/source-layer-c/libcxx/src/ios.cpp	2026-08-05 11:41:57.705122452 +0800
+++ tmp/R53/source-layer-c-full/libcxx/src/ios.cpp	2026-08-20 22:49:04.183298384 +0800
@@ -9,6 +9,7 @@
 #include <__config>
 #include <__locale>
 #include <algorithm>
+#include <cxxabi.h>
 #include <ios>
 #include <limits>
 #include <memory>
@@ -362,6 +363,12 @@
 void ios_base::__set_badbit_and_consider_rethrow() {
   __rdstate_ |= badbit;
 #if _LIBCPP_HAS_EXCEPTIONS
+  try {
+    throw;
+  } catch (const __cxxabiv1::__forced_unwind&) {
+    throw;
+  } catch (...) {
+  }
   if (__exceptions_ & badbit)
     throw;
 #endif // _LIBCPP_HAS_EXCEPTIONS
@@ -370,6 +377,12 @@
 void ios_base::__set_failbit_and_consider_rethrow() {
   __rdstate_ |= failbit;
 #if _LIBCPP_HAS_EXCEPTIONS
+  try {
+    throw;
+  } catch (const __cxxabiv1::__forced_unwind&) {
+    throw;
+  } catch (...) {
+  }
   if (__exceptions_ & failbit)
     throw;
 #endif // _LIBCPP_HAS_EXCEPTIONS
