--- baseline
+++ candidate
@@ -202,6 +202,7 @@
     -DCMAKE_C_FLAGS="$CFLAGS" \
     -DCMAKE_CXX_FLAGS="$CXXFLAGS" \
     -DLLVM_ENABLE_LIBCXX=ON \
+    -DLIBCLANG_BUILD_STATIC=ON \
     -DCMAKE_SHARED_LINKER_FLAGS="-fuse-ld=lld -ffunction-sections -fdata-sections -Wl,--gc-sections -Wl,--no-as-needed -lc++abi -Wl,--as-needed" \
     -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=lld -ffunction-sections -fdata-sections -Wl,--gc-sections -Wl,--no-as-needed -lc++abi -Wl,--as-needed" \
     -DLLVM_USE_LINKER=lld \
