--- /home/toolchain/development/libc++_replacement/tmp/GBS-ROOT/LIBCXX-2218-armv7l-20260805-r3b/local/BUILD-ROOTS/scratch.armv7l.0/usr/bin/ldd	2026-07-22 18:35:39.000000000 +0800
+++ /home/toolchain/development/libc++_replacement/progress/T4/probes/ldd.armv7l.qemu-static	2026-08-05 15:29:20.329498764 +0800
@@ -26,7 +26,7 @@
 TEXTDOMAIN=libc
 TEXTDOMAINDIR=/usr/share/locale
 
-RTLDLIST=/lib/ld-linux.so.3
+RTLDLIST=/home/toolchain/development/libc++_replacement/progress/T4/probes/rtld-armv7l-qemu-wrapper.bin
 warn=
 bind_now=
 verbose=
