--- codes/R104/bcc-tools:5bd0654e24bb231035ea684d278f674aa95a2cd2:packaging/bcc-tools.spec
+++ tmp/GBS-ROOT/BUILD-W2-bcc-x86-libcxx/local/sources/tizen_base_standard/bcc-tools-0.35.0-0/bcc-tools.spec
@@ -7,6 +7,7 @@
 %endif
 
 Name: bcc-tools
+VCS:  platform/upstream/bcc#dd26ba03fc8109396bf841f1bbf7b4df0a8ed9d2-dirty
 Version: 0.35.0
 Release: 0
 Summary: Tools for BPF-based Linux IO analysis, networking, monitoring
