--- codes/R104/abseil-cpp:7a4f04dad6d30ba09c406837cdd23ad779a69210:packaging/abseil-cpp.spec
+++ tmp/GBS-ROOT/R104-matrix-aarch64-libcxx/local/sources/tizen_unified_standard/abseil-cpp-20250814.1-0/abseil-cpp.spec
@@ -22,6 +22,7 @@
 %endif
 
 Name:           abseil-cpp
+VCS:            platform/upstream/abseil-cpp#9c39c516dca54c786ddc5da45a5a15acf41ef75e-dirty
 Version:        20250814.1
 Release:        0
 Summary:        C++11 libraries which augment the C++ stdlib
