--- codes/R104/jsoncpp:9cbab147f44097d5b8da2b3ef38f75376a282d01:packaging/jsoncpp.spec
+++ tmp/GBS-ROOT/R104-matrix-armv7l-libcxx/local/sources/tizen_unified_standard/jsoncpp-1.9.6-0/jsoncpp.spec
@@ -7,6 +7,7 @@
 %endif
 
 Name:       jsoncpp
+VCS:        platform/upstream/jsoncpp#96bf0c36eed62a6d7bfc9bdb30f0e41d877aa6f0-dirty
 Version:    1.9.6
 Release:    0
 Summary:    JSON library implemented in C++
