--- codes/R104/taglib:8b552bdf27c913a93bca664c41c3fafb214962a1:packaging/taglib.spec
+++ tmp/GBS-ROOT/R104-matrix-armv7l-libcxx/local/sources/tizen_unified_standard/taglib-1.13.1-1/taglib.spec
@@ -5,6 +5,7 @@
 %endif
 
 Name:      taglib
+VCS:       platform/upstream/taglib#4e7de646d897ed5a92111799058b21705a3bd34c-dirty
 Summary:   library for reading and editing the meta-data of several popular audio formats
 Version:   1.13.1
 Release:   1
