--- codes/R104/boost:4168e873584ae50a12733262026c007dce18ec89:packaging/boost.spec
+++ tmp/GBS-ROOT/R105-boost-aarch64/local/sources/tizen_unified_standard/boost-1.83.0-0/boost.spec
@@ -57,6 +57,7 @@
 %define all_libs %{most_libs} %{?context_libs} %{?fiber_libs}
 
 Name:           boost
+VCS:            platform/upstream/boost#69203710b8582c489f420ef53a769cc6077c250b-dirty
 Url:            http://www.boost.org
 Summary:        Boost C++ Libraries
 License:        BSL-1.0
