--- codes/libcynara-commons/packaging/libcynara-commons.spec
+++ tmp/GBS-ROOT/R30-baseline-libcynara-armv7l-20260812-05/local/sources/r30_libcynara_default_armv7l/libcynara-commons-0.26.0-1/libcynara-commons.spec
@@ -1,4 +1,5 @@
 Name:       libcynara-commons
+VCS:        git/platform/core/security/libcynara-commons#4b60376089951e8e379cc7a23f7509f433a67525
 Summary:    Cynara - cynara commons library
 Version:    0.26.0
 Release:    1
