--- codes/security-manager/packaging/security-manager.spec
+++ tmp/GBS-ROOT/R30-baseline-security-armv7l-20260812-01/local/sources/r30/security-manager-2.4.0-0/security-manager.spec
@@ -1,6 +1,7 @@
 %{!?build_type:%global build_type RELEASE}
 
 Name:       security-manager
+VCS:        git/platform/core/security/security-manager#f6a98bfdbf86f48a66feb81e81e4c8b488c9a550-dirty
 Summary:    Security manager and utilities
 Version:    2.4.0
 Release:    0
