--- codes/libcynara-commons/packaging/libcynara-commons.spec
+++ tmp/GBS-ROOT/R30-patched-libcynara-default-armv7l-20260812-01/local/sources/r30_libcynara_default_armv7l/libcynara-commons-0.26.0-1/libcynara-commons.spec
@@ -1,4 +1,5 @@
 Name:       libcynara-commons
+VCS:        git/platform/core/security/libcynara-commons#5ba41186b38682290575ceff60143c9009e531f7
 Summary:    Cynara - cynara commons library
 Version:    0.26.0
 Release:    1
