--- codes/security-manager/packaging/security-manager.spec
+++ tmp/GBS-ROOT/R30-patched-security-default-armv7l-20260812-01/local/sources/r30_default_armv7l/security-manager-2.4.0-0/security-manager.spec
@@ -1,6 +1,7 @@
 %{!?build_type:%global build_type RELEASE}
 
 Name:       security-manager
+VCS:        git/platform/core/security/security-manager#dffe51106564d8db0a40b6ff7c01d6fcece816af
 Summary:    Security manager and utilities
 Version:    2.4.0
 Release:    0
