--- tmp/BUILD_NIGHT_0916.h4taGY/bpftrace-push:7831fb34c6881b5cff0b8bdd0dac27a1829290fa:packaging/bpftrace.spec
+++ tmp/WEEKEND_0918/specs/bpftrace.spec
@@ -13,7 +13,6 @@
 Source0:        %{name}-%{version}.tar.gz
 Source1001:     %{name}.manifest
 Source1002:     prebuilt-static-bpftrace.tar.gz
-Patch0:         static-link-libbpf.patch
 
 ExclusiveArch:  armv7l aarch64
 
@@ -76,7 +75,6 @@
 
 %prep
 %setup -q
-%patch0 -p1
 cp %{SOURCE1001} .
 tar xzvf %{SOURCE1002}
 %if 0%{?build_with_libcxx}
