--- codes/llvm/libcxxabi/include/cxxabi.h	2026-08-05 11:41:57.984830359 +0800
+++ tmp/R33/llvm-exp/libcxxabi/include/cxxabi.h	2026-08-13 22:29:58.003731425 +0800
@@ -37,6 +37,9 @@
 // runtime routines use C calling conventions, but are in __cxxabiv1 namespace
 namespace __cxxabiv1 {
 
+// R33 experiment only: a type token used to match forced unwinding.
+class _LIBCXXABI_TYPE_VIS __forced_unwind {};
+
 struct __cxa_exception;
 
 extern "C"  {
