--- /home/toolchain/development/libcxx_runtime_validation/codes/llvm/libcxx/include/__condition_variable/condition_variable.h	2026-08-05 11:41:57.639876670 +0800
+++ /home/toolchain/development/libcxx_runtime_validation/tmp/R115-W4/control/include/__condition_variable/condition_variable.h	2026-09-07 21:36:36.909007553 +0800
@@ -104,7 +104,7 @@
   void notify_one() _NOEXCEPT;
   void notify_all() _NOEXCEPT;
 
-  void wait(unique_lock<mutex>& __lk) _NOEXCEPT;
+  void wait(unique_lock<mutex>& __lk);
 
   template <class _Predicate>
   _LIBCPP_HIDE_FROM_ABI void wait(unique_lock<mutex>& __lk, _Predicate __pred) {
@@ -174,14 +174,14 @@
 
 private:
   void
-  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<chrono::system_clock, chrono::nanoseconds>) _NOEXCEPT;
+  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<chrono::system_clock, chrono::nanoseconds>);
 #  if _LIBCPP_HAS_COND_CLOCKWAIT
   _LIBCPP_HIDE_FROM_ABI void
-  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<chrono::steady_clock, chrono::nanoseconds>) _NOEXCEPT;
+  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<chrono::steady_clock, chrono::nanoseconds>);
 #  endif
   template <class _Clock>
   _LIBCPP_HIDE_FROM_ABI void
-  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<_Clock, chrono::nanoseconds>) _NOEXCEPT;
+  __do_timed_wait(unique_lock<mutex>& __lk, chrono::time_point<_Clock, chrono::nanoseconds>);
 };
 #endif // _LIBCPP_HAS_THREADS
 
@@ -195,7 +195,7 @@
 
 #  if _LIBCPP_HAS_COND_CLOCKWAIT
 inline void condition_variable::__do_timed_wait(
-    unique_lock<mutex>& __lk, chrono::time_point<chrono::steady_clock, chrono::nanoseconds> __tp) _NOEXCEPT {
+    unique_lock<mutex>& __lk, chrono::time_point<chrono::steady_clock, chrono::nanoseconds> __tp) {
   using namespace chrono;
   if (!__lk.owns_lock())
     std::__throw_system_error(EPERM, "condition_variable::timed wait: mutex not locked");
@@ -219,7 +219,7 @@
 
 template <class _Clock>
 inline void condition_variable::__do_timed_wait(unique_lock<mutex>& __lk,
-                                                chrono::time_point<_Clock, chrono::nanoseconds> __tp) _NOEXCEPT {
+                                                chrono::time_point<_Clock, chrono::nanoseconds> __tp) {
   wait_for(__lk, __tp - _Clock::now());
 }
 
