--- /home/toolchain/development/libcxx_runtime_validation/codes/llvm/libcxx/src/condition_variable.cpp	2026-08-05 11:41:57.701719952 +0800
+++ /home/toolchain/development/libcxx_runtime_validation/tmp/R115-W4/condition_variable_removed.cpp	2026-09-07 21:36:36.911188465 +0800
@@ -30,7 +30,7 @@
 
 void condition_variable::notify_all() noexcept { __libcpp_condvar_broadcast(&__cv_); }
 
-void condition_variable::wait(unique_lock<mutex>& lk) noexcept {
+void condition_variable::wait(unique_lock<mutex>& lk) {
   if (!lk.owns_lock())
     std::__throw_system_error(EPERM, "condition_variable::wait: mutex not locked");
   int ec = __libcpp_condvar_wait(&__cv_, lk.mutex()->native_handle());
@@ -39,7 +39,7 @@
 }
 
 void condition_variable::__do_timed_wait(unique_lock<mutex>& lk,
-                                         chrono::time_point<chrono::system_clock, chrono::nanoseconds> tp) noexcept {
+                                         chrono::time_point<chrono::system_clock, chrono::nanoseconds> tp) {
   using namespace chrono;
   if (!lk.owns_lock())
     std::__throw_system_error(EPERM, "condition_variable::timed wait: mutex not locked");
