--- a/src/ast/CMakeLists.txt
+++ b/src/ast/CMakeLists.txt
@@ -83,7 +83,7 @@
   if(LIBCLANG_STATIC_PATH)
     get_filename_component(LIBCLANG_STATIC_ABS_PATH "${CMAKE_SOURCE_DIR}/${LIBCLANG_STATIC_PATH}" ABSOLUTE)
     message(STATUS "Using custom libclang absolute path: ${LIBCLANG_STATIC_ABS_PATH}")
-    target_link_libraries(ast PUBLIC ${LIBCLANG_STATIC_ABS_PATH} bpf clangIndex clangDriver clangFrontend clangCodeGen)
+    target_link_libraries(ast PUBLIC ${LIBCLANG_STATIC_ABS_PATH} ${LIBBPF_LIBRARIES} clangIndex clangDriver clangFrontend clangCodeGen)
   else()
     target_link_libraries(ast PUBLIC libclang_static clangDriver clangFrontend clangCodeGen)
   endif()
