--- baseline
+++ candidate
@@ -1,3 +1,15 @@
+
+
+
+
+
+
+
+
+
+
+
+
 
 
 
@@ -91,17 +103,17 @@
   -DLLVM_INCLUDE_TESTS=OFF \
   -DLIBCXX_ENABLE_SHARED=ON \
   -DLIBCXX_ENABLE_TIZEN_CONDVAR_VERSIONS=ON \
-  -DLIBCXX_ENABLE_STATIC=OFF \
+  -DLIBCXX_ENABLE_STATIC=ON \
   -DLIBCXX_INSTALL_SHARED_LIBRARY=ON \
-  -DLIBCXX_INSTALL_STATIC_LIBRARY=OFF \
+  -DLIBCXX_INSTALL_STATIC_LIBRARY=ON \
   -DLIBCXX_CXX_ABI=libcxxabi \
   -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=OFF \
   -DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=ON \
   -DLIBCXX_USE_COMPILER_RT=OFF \
   -DLIBCXXABI_ENABLE_SHARED=ON \
-  -DLIBCXXABI_ENABLE_STATIC=OFF \
+  -DLIBCXXABI_ENABLE_STATIC=ON \
   -DLIBCXXABI_INSTALL_SHARED_LIBRARY=ON \
-  -DLIBCXXABI_INSTALL_STATIC_LIBRARY=OFF \
+  -DLIBCXXABI_INSTALL_STATIC_LIBRARY=ON \
   -DLIBCXXABI_USE_COMPILER_RT=OFF \
   -DLIBCXXABI_USE_LLVM_UNWINDER=OFF \
   -DLIBCXXABI_ENABLE_STATIC_UNWINDER=OFF \
@@ -139,6 +151,7 @@
 /usr/include/c++/v1
 /usr/lib/libc++.so
 /usr/lib/libc++experimental.a
+/usr/lib/libc++.a
 /usr/lib/libc++.modules.json
 /usr/share/libc++/v1
 /usr/share/doc/libcxx-runtimes/SOURCE_PROVENANCE
@@ -153,6 +166,7 @@
 %manifest libcxx-runtimes.manifest
 %defattr(-,root,root,-)
 /usr/lib/libc++abi.so
+/usr/lib/libc++abi.a
 
 %changelog
 * Tue Aug 04 2026 Toolchain ABI Team <toolchain@example.invalid> 22.1.8-1
